docs/developers/bitesize/workflowExecution/executionAndValueStoreExamples.md

```diff
@@ -0,0 +1,229 @@
+# Workflow Execution: Execution Store and Value Store Examples
+
+## Introduction
+
+This page provides run-throughs to give insight into how the
+[Execution Store](executionStore.md) and [Value Store](valueStore.md) work in
+practice in some example situations. 
+
+## Handling a single task call
+
+To begin, consider this simple workflow. It has a single task call whose
+result is exposed as an output String:
+
+```wdl
+version 1.0
+
+workflow single_task_workflow {
+  call single_task
+  
+  output {
+    String string_out = single_task.string_out
+  }
+}
+
+task single_task {
+  command {
+    echo hello
+  }
+  output {
+    String string_out = "hello"
+  }
+}
+```
+
+The **Execution Store** will keep track of statuses as the workflow runs:
+
+| | `single_task` | `string_out` |
+|---|---|---|
+|1|`NotStarted`|`NotStarted`|
+|2|`QueuedInCromwell`|`NotStarted`|
+|3|`Starting`|`NotStarted`|
+|4|`Running`|`NotStarted`|
+|5|`Done`|`NotStarted`|
+|6|`Done`|`Running`|
+|7|`Done`|`Done`|
+
+In step 1, the workflow has just started and the ExecutionStore is created in its initial
+state. The Value Store doesn't track statuses and so begins empty: `{ }`.
+
+In steps 2-4, the Execution Store tracks the `single_task` job as the engine is executing it.
+
+As the Execution Store is updated to indicate task completion is step 5, the Value Store is also updated to
+include the output value of the task:
+```json
+{
+  "single_task.string_out": "hello"
+}
+```
+
+By step 6, Cromwell can use the fact that the task is complete to decide that the output node is ready to be
+evaluated. And the input to the output expression is available for lookup in the Value Store.
+
+In step 7, all workflow nodes have run and the workflow is complete. The Value Store is updated once again to
+additionally contain the output node value:
+```json
+{
+  "single_task.string_out": "hello",
+  "string_out": "hello"
+}
+``` 
+
+Cromwell can use this information to trigger the "workflow complete" logic.
+
+## Handling scatters
+
+When we run scattered tasks, the Execution Store cannot tell ahead of time how many
```

Comment:
```suggestion
When Cromwell runs scattered tasks, the Execution Store cannot tell ahead of time how many
```
[ToL] I think a detached/impersonal voice works better for docs (here and elsewhere)

Reply:
Kate would be proud! I'll do my best to reword these